src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -111,11 +112,24 @@
     private transient Project gitlabProject;
     private int projectId = -1;
 
+    /**
+     * The cache of {@link ObjectMetadataAction} instances for each open MR.
+     */
+    @NonNull
+    private transient /*effectively final*/ Map<Integer,ObjectMetadataAction> mergeRequestMetadataCache;
+    /**
+     * The cache of {@link ObjectMetadataAction} instances for each open MR.
+     */
+    @NonNull
+    private transient /*effectively final*/ Map<Integer,ContributorMetadataAction> mergeRequestContributorCache;
+
     @DataBoundConstructor
     public GitLabSCMSource(String serverName, String projectOwner, String projectPath) {
         this.serverName = serverName;
         this.projectOwner = projectOwner;
         this.projectPath = projectPath;
+        mergeRequestMetadataCache = new HashMap<>();
```

Comment:
No, use ConcurrentHashMap and need to initialize in `readResolve`.
You could also initialize on the field itself. Which avoids the need for initializing in constructor and `readResolve`
```suggestion
        mergeRequestMetadataCache = new ConcurrentHashMap<>();
```

Reply:
I am wondering why we need concurrentHashMap over HashMap?

Reply:
You can have multiple events adding to the map

Reply:
or retrieving from it